### ISSUE TITLE
MissingOverrideAttribute false positive with traits used by child classes

### ISSUE THREAD

**Issue by pilif** [vimeo/psalm#10982]
https://psalm.dev/r/70ba038824

If you do what psalm says you're supposed to do and add the `Override` attribute to the trait method, you will create invalid code that PHP refuses to run (and psalm also flags as an error).

I think `B` should be totally free to import its own copy of the `Foo` trait.

**Comment by ghostwriter** [comment]
- Private methods of a parent class do not satisfy #[\Override], because it's not part of the externally visible API.

**Comment by pilif** [comment]
indeed. Hence why psalm shouldn't ask me to add `#[\Override]` hence my bug report.

**Comment by whataboutpereira** [comment]
Still an issue.

https://psalm.dev/r/02a66689aa

If you fix the code as per suggestion you will end up with `Fatal error: B::inTrait() has #[\Override] attribute, but no matching parent method exists`
